.github/workflows/qhelp-pr-preview.yml

```
@@ -52,12 +71,26 @@ jobs:
                echo '```'
             fi
             echo "</details>"
-          done < "${RUNNER_TEMP}/paths.txt" >> comment.txt
+          done < "${RUNNER_TEMP}/paths.txt" >> comment_body.txt
           exit "${EXIT_CODE}"
 
       - if: always()
         uses: actions/upload-artifact@v2
         with:
           name: comment
-          path: comment.txt
+          path: comment_body.txt
+          retention-days: 1
+
+      - name: Save ID of existing QHelp comment (if it exists)
+        run: |
+          # Find the latest comment starting with "QHelp previews"
+          COMMENT_PREFIX="QHelp previews"
+          gh api "repos/${{ github.repository }}/issues/${{ github.event.number }}/comments" --paginate --jq "[.[] | select(.body|startswith(\"${COMMENT_PREFIX}\")) | .id] | max" > comment_id.txt
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
+
+      - uses: actions/upload-artifact@v2
+        with:
+          name: comment
+          path: comment_id.txt
```

Shall we add a `if-no-files-found:` field to document that it expected for the `comment_id.txt` to be missing? Perhaps we should add `if-no-files-found: error` to the upload steps for comment_body and pr_number.
```suggestion
          path: comment_id.txt
          if-no-files-found: ignore
```

---

Actually due to the issue I mentioned [here](https://github.com/github/codeql/pull/8618#discussion_r842273725), we should always upload a `comment_id.txt` file regardless, so I've made them all `if-no-files-found: error`.